src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php

```diff
@@ -85,12 +86,54 @@ protected function setup_template_vars() {
 		$template_vars = parent::setup_template_vars();
 
 		// Here update, add and remove from the default template vars.
-		$template_vars['view_more_link']   = tribe_get_events_link();
-		$template_vars['widget_title']     = $this->context->get( 'widget_title' );
-		$template_vars['show_latest_past'] = false;
+		$template_vars['view_more_link']     = tribe_get_events_link();
+		$template_vars['widget_title']       = $this->context->get( 'widget_title' );
+		$template_vars['no_upcoming_events'] = $this->context->get( 'no_upcoming_events' );
+		$template_vars['show_latest_past']   = false;
 		// Display is modified with filters in Pro.
 		$template_vars['display'] = [];
 
 		return $template_vars;
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function setup_messages( array $events ) {
+		if ( ! empty( $events )	) {
+			return;
+		}
+
+		$keyword = $this->context->get( 'keyword', false );
+		$this->messages->insert(
+			Messages::TYPE_NOTICE,
+			Messages::for_key( 'no_upcoming_events', trim( $keyword ) )
+		);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function setup_the_loop( array $args = [] ) {
+		global $wp_query;
+
+		$this->global_backup = [
+			'wp_query'   => $wp_query,
```

Comment:
:warning: Array double arrow not aligned correctly; expected 1 space(s) between "'wp_query'" and double arrow, but found 3.

<sub>hash: 9db522f73013781b830564a1026e97b7</sub>